fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -1789,6 +1791,45 @@ private LoanCharge retrieveLoanChargeBy(final Long loanId, final Long loanCharge
         return loanCharge;
     }
 
+	@Override
+    @CronTarget(jobName = JobName.ALLOCATE_OVERPAYMENTS_TO_SAVINGS)
+    public void allocateOverpayments() throws JobExecutionException {
+        final Collection<Long> loanIds = this.loanReadPlatformService.fetchOverpayedLoansForAllocation();
+        final StringBuilder sb = new StringBuilder();
+
+        if(loanIds != null){
+            for(final Long loanId : loanIds){
+
+                try{
+                    final AccountAssociations accountAssociations = this.accountAssociationRepository.findByLoanIdAndType(loanId,AccountAssociationType.LINKED_ACCOUNT_ASSOCIATION.getValue());
```

Comment:
Assume a case with over paid loans which doesn't have linked savings account. That time accountAssociations will be null right?